app/src/main/java/com/m/services/dataSync/TaskActivity.java

```diff
@@ -106,4 +105,33 @@ private void updateUi(){
         runOnUiThread(() -> itemAdapter.notifyDataSetChanged());
     }
 
+    private void uiToast(String message) {
+        runOnUiThread(() -> Toast.makeText(getApplicationContext(), message, Toast.LENGTH_LONG).show());
+
+    }
+
+    private void handleErrorExceptions(ApolloException error){
+        if (error instanceof ApolloHttpException){
+            handleHttpErrors((ApolloHttpException) error);
+        } else if (error instanceof ApolloNetworkException){
+            uiToast(error.getMessage());
+        } else {
+            uiToast(error.getMessage());
+        }
+    }
+
+    private void handleHttpErrors(ApolloHttpException error){
```

Comment:
just as an alternative, I found out that we can filter through different types of errors in the query/mutation/subscription with: 
```
@Override
                             public void onFailure(@NotNull ApolloException e) {
                                 System.out.println("FAILURE " + e);
                             }

                             @Override
                             public void onHttpError(@NotNull ApolloHttpException e) {
                                 System.out.println("HttpError " + e);
                             }

                             @Override
                             public void onNetworkError(@NotNull ApolloNetworkException e) {
                                 System.out.println("Network error " + e);
                             }

                             @Override
                             public void onParseError(@NotNull ApolloParseException e) {
                                 System.out.println("Parse Error " + e);
                             }
                         });
```